src/command/verify/verify.c

```
@@ -1661,8 +1676,19 @@ verifyProcess(const bool verboseText)
                     .expression = backupRegExpStr),
                 sortOrderAsc);
 
-            if (backupLabel != NULL && strLstEmpty(jobData.backupList))
-                THROW_FMT(BackupSetInvalidError, "backup set %s is not valid", strZ(backupLabel));
+            if (!backupLabelInvalid && backupLabel != NULL && strLstEmpty(jobData.backupList))
```

Now if the label is not valid, we will still do a directory search.

---

Okay, another idea: we can do ternary expression for backupList and archiveList (replacing them with empty lists if backupLabel is invalid), and then we can also revert this condition and the condition on line 1701. How about that?

---

Maybe. Unless it's probably better to use a regular if. I think it will be more readable. One more point, should we check the archives if we haven't found a backup?

---

Got it. No, I think we shouldn't check them: `--set` is used when the user tries to check a specific backup. Also, checking all archives is slow, and I think we should instead report the error immediately.

---

[Inside cmd Info](https://github.com/arenadata/pgbackrest/blob/9546b9c5d0e4da7bd952b01cc3374c26e9f90ea0/src/command/info/info.c#L1283) they don't read archive if backup didn't found

---

Sorry, it's about *.info files, not this case